### ISSUE TITLE
Cannot clone a model

### ISSUE THREAD

**Issue by ThisIsMonta** [ThatOpen/engine_components#439]
### Describe the bug 📝

Trying to clone a model but got an error : 

```
@thatopen_components.js?v=6f57743e:92 Uncaught (in promise) 
TypeError: Cannot read properties of undefined (reading 'index')
    at new Js (@thatopen_components.js?v=6f57743e:92:62)
    at Js.clone (chunk-NSW2P6WQ.js?v=6f57743e:5400:12)
    at Gi.copy (chunk-NSW2P6WQ.js?v=6f57743e:5425:24)
    at Gi.clone (chunk-NSW2P6WQ.js?v=6f57743e:5400:35)
    at ifc_worker.js?worker…e&type=module:49:24
```

<img width="595" alt="Screen Shot 2024-07-03 at 14 23 03" src="https://github.com/ThatOpen/engine_components/assets/47502752/21e19d81-9bbd-4528-8836-2de6c91c8bf7">

### System Info 💻

```shell
.
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/ThatOpen/engine_components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a repository issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by agviegas** [comment]
Related to #222. Probably both issues can be solved at the same time.

**Comment by agviegas** [comment]
We've implemented fragment cloning (and it can also be used to create fragment subsets). 

It's available from `@thatopen/components@2.1.17` and `@thatopen/fragments@2.1.6`. By modifying the [IFCLoader tutorial](https://docs.thatopen.com/Tutorials/Components/Core/IfcLoader) code like this:

```ts
async function loadIfc() {
  const file = await fetch(
    "https://thatopen.github.io/engine_components/resources/small.ifc",
  );
  const data = await file.arrayBuffer();
  const buffer = new Uint8Array(data);
  const model = await fragmentIfcLoader.load(buffer);
  model.name = "example";
  world.scene.three.add(model);

  const classifier = components.get(OBC.Classifier);
  classifier.byEntity(model);
  const found = classifier.find({
    entities: ["IFCWALLSTANDARDCASE", "IFCSLAB"],
  });

  const clone = fragments.clone(model, found);
  world.scene.three.add(clone);
  window.addEventListener("keydown", (e) => {
    clone.position.x += 1;
  });

  const map = clone.getFragmentMap([186]);
  const hider = components.get(OBC.Hider);
  hider.set(false, map);
}
```

I get this:

https://github.com/user-attachments/assets/8fc61ff2-d9b3-4c85-aaa7-eef65ffa3de7
